src/main/java/org/broadinstitute/dsm/model/BSPKit.java

```diff
@@ -0,0 +1,150 @@
+package org.broadinstitute.dsm.model;
+
+import lombok.NonNull;
+import org.apache.commons.lang3.StringUtils;
+import org.broadinstitute.ddp.db.TransactionWrapper;
+import org.broadinstitute.dsm.db.DDPInstance;
+import org.broadinstitute.dsm.db.InstanceSettings;
+import org.broadinstitute.dsm.db.dao.ddp.kitrequest.KitRequestDao;
+import org.broadinstitute.dsm.db.dto.settings.InstanceSettingsDto;
+import org.broadinstitute.dsm.model.bsp.BSPKitInfo;
+import org.broadinstitute.dsm.model.bsp.BSPKitQueryResult;
+import org.broadinstitute.dsm.model.bsp.BSPKitStatus;
+import org.broadinstitute.dsm.statics.ApplicationConfigConstants;
+import org.broadinstitute.dsm.statics.ESObjectConstants;
+import org.broadinstitute.dsm.util.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.sql.Connection;
+import java.util.HashMap;
+import java.util.Map;
+
+public class BSPKit {
+    private static Logger logger = LoggerFactory.getLogger(BSPKit.class);
+
+    public BSPKitStatus getKitStatus(@NonNull String kitLabel, NotificationUtil notificationUtil) {
+        logger.info("Checking label " + kitLabel);
+        BSPKitQueryResult bspKitQueryResult = BSPKitQueryResult.getBSPKitQueryResult(kitLabel);
+
+        if (bspKitQueryResult == null) {
+            logger.info("No kit w/ label " + kitLabel + " found");
+            return null;
+        }
+
+        if (StringUtils.isNotBlank(bspKitQueryResult.getParticipantExitId())) {
+            String message = "Kit of exited participant " + bspKitQueryResult.getBspParticipantId() + " was received by GP.<br>";
+            notificationUtil.sentNotification(bspKitQueryResult.getNotificationRecipient(), message, NotificationUtil.DSM_SUBJECT);
+            return new BSPKitStatus(BSPKitStatus.EXITED);
+        }
+        if (StringUtils.isNotBlank(bspKitQueryResult.getDeactivationDate())) {
+            return new BSPKitStatus(BSPKitStatus.DEACTIVATED);
+        }
+        return null;
+    }
+
+    public boolean canReceiveKit(@NonNull String kitLabel) {
+        BSPKitQueryResult bspKitQueryResult = BSPKitQueryResult.getBSPKitQueryResult(kitLabel);
+        if (bspKitQueryResult == null) {
+            logger.info("No kit w/ label " + kitLabel + " found");
+            return false;
+        }
+        if (StringUtils.isNotBlank(bspKitQueryResult.getParticipantExitId()) || StringUtils.isNotBlank(bspKitQueryResult.getDeactivationDate())) {
+            logger.info("Kit can not be received.");
+            return false;
+        }
+        return true;
+
+    }
+
+    public BSPKitInfo receiveBSPKit(String kitLabel, NotificationUtil notificationUtil) {
```

Comment:
too many things coupled to each other in one method which makes it so complex and confusing, this complexity could be cut by removing `TransactionWrapper.inTransaction`, since `BSPKit` is business logic class it is bad idea to have direct connection to database from it, instead it is better to use composition with `Dao/Dto` classes.